include/zephyr/posix/sys/utsname.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_POSIX_SYS_UTSNAME_H_
+#define ZEPHYR_INCLUDE_POSIX_SYS_UTSNAME_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct utsname {
+	char sysname[sizeof("Zephyr")];
+	char nodename[CONFIG_POSIX_UNAME_NODENAME_LEN];
+	char release[sizeof("99.99.99")];
+	char version[CONFIG_POSIX_UNAME_VERSION_LEN];
```

Comment:
I think the size of this field can also be computed -- just need to insert the computation of the version string into this header. One less config value to deal with seems like a nice thing.

Reply:
We can actually compute everything using preprocessor defines here, at the expense of including 2 more headers & some macro magic then looks a bit complicated

Reply:
The version field can be fixed size as well, technically, too, if we omit the date and time, and it's compatible with the banner version as well.

```cpp
char version[sizeof("zephyr-v99.99.99-9999-g0123456789ab")];
```

Ideally we could just include `<version.h>`, which is generated at build time and use the exact `#define`s that appear there.

I'm hesitant to do that, just because it really isn't referenced outside of `banner.c`, which gives me the impression that it's private somehow.

If that makes life easier though, by all means, just 
```cpp
struct utsname {
	char sysname[sizeof("Zephyr")];
	char nodename[CONFIG_POSIX_UNAME_NODENAME_LEN + 1];
	char release[sizeof(KERNEL_VERSION_STRING)];
	char version[sizeof(STRINGIFY(BUILD_VERSION))];
	char machine[sizeof(CONFIG_ARCH)];
};
```

And honestly, the `__DATE__` and `__TIME__` macros are probably not *super* critical.

Reply:
`__DATE__` and `__TIME__` break build reproducibility and should not be enabled by default (not sure what is the current situation)